cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/SourceControlMeta.java

```diff
@@ -44,18 +49,30 @@ public boolean equals(Object o) {
 
     SourceControlMeta that = (SourceControlMeta) o;
 
-    return Objects.equals(fileHash, that.fileHash);
+    return Objects.equals(fileHash, that.fileHash)
+        && Objects.equals(commitId, that.commitId)
+        && Objects.equals(lastSyncedAt, that.lastSyncedAt);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(fileHash);
+    return Objects.hash(fileHash, lastSyncedAt, commitId);
   }
 
   @Override
   public String toString() {
     return "SourceControlMeta{"
         + "fileHash='" + fileHash + '\''
+        + "commitId='" + commitId + '\''
+        + "lastSyncedAt='" + lastSyncedAt + '\''
         + '}';
   }
+
+  public String getCommitId() {
+    return commitId;
+  }
+
+  public Instant getLastSyncedAt() {
```

Comment:
would be good to add a javadoc, since 'sync' can be a bit ambiguous

Reply:
done

Reply:
let's be sure a null sync time is tested in the UI too. This type of thing (adding a new field) has caused regressions in the past.